implementations/micrometer-registry-otlp/src/main/java/io/micrometer/registry/otlp/OtlpMetricsSender.java

```diff
@@ -23,17 +27,125 @@
  *
  * @since 1.15.0
  */
-public interface OtlpMetricsSender extends MetricsSender {
+public interface OtlpMetricsSender {
 
     /**
      * Send a batch of OTLP Protobuf format metrics to an OTLP receiver.
-     * @param address address of the OTLP receiver to which metrics will be sent
-     * @param metricsData OTLP protobuf encoded batch of metrics
-     * @param headers metadata to send as headers with the metrics data
-     * @throws Throwable when there is an exception in sending the metrics; the caller
+     * @param request metrics request to publish
+     * @throws Exception when there is an exception in sending the metrics; the caller
      * should handle this in some way such as logging the exception
      */
-    @Override
-    void send(String address, byte[] metricsData, Map<String, String> headers) throws Throwable;
+    void send(Request request) throws Exception;
+
+    /**
+     * Convenience method for building a request to use with this sender.
+     * @param metricsData metrics data that will be sent
+     * @return builder
+     * @see Request.Builder#send
+     */
+    default Request.Builder newRequest(byte[] metricsData) {
+        return new Request.Builder(metricsData, this);
+    }
+
+    /**
+     * Immutable representation of a payload of metrics to use with a
+     * {@link OtlpMetricsSender}.
+     *
+     * @since 1.15.0
+     */
+    class Request {
+
+        @Nullable
+        private final String address;
+
+        private final Map<String, String> headers;
+
+        private final byte[] metricsData;
+
+        /**
+         * Represents a payload of metrics to be sent.
+         * @param address where to send the metrics
+         * @param metricsData encoded batch of metrics
+         * @param headers metadata to send as headers with the metrics data
+         */
+        public Request(@Nullable String address, Map<String, String> headers, byte[] metricsData) {
```

Comment:
When using a builder consider making the constructor private. This way there's only one way of using the API

Reply:
I've made it private now. We can always make the scope more public later if it makes sense, so it's probably best to keep it more closed now, especially since this is API that most people won't touch (only those making their own OtlpMeterRegistry implementation of `publish` etc will need to use this)